### PR TITLE
fix(Ludicrous): Upserts on list type in Dgraph

### PR DESCRIPTION
**Pull request by rahulgurnani**
Fixes DGRAPH-2446

We were not updating startTs for upsert mutations properly in ludicrous mode.

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/6754)

### CODE REVIEW

File: systest/ludicrous/upsert_test.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	require.NoError(t, err)
+	var dat ResponseData
+	err = json.Unmarshal(res.Json, &dat)
+	require.NoError(t, err)
+
+	require.Equal(t, len(dat.All[0].Counts), 10)
+}
+
+func TestSequentialUpdate(t *testing.T) {
+	t.Log("TestSequentialUpdate")
```
Comment (jarifibrahim) [inline systest/ludicrous/upsert_test.go:110]: Remove this please.

File: systest/ludicrous/upsert_test.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	All []Data `json:"all,omitempty"`
+}
+
+func InitData(t *testing.T) {
+	dg, err := testutil.DgraphClient(testutil.SockAddr)
+	testutil.DropAll(t, dg)
+	require.NoError(t, err)
+	schema := `
+		name: string @index(exact) .
+		email: string @index(exact) .
```
Comment (jarifibrahim) [inline systest/ludicrous/upsert_test.go:35]: You can remove the email. It seems like we're not using it anywhere.

File: systest/ludicrous/upsert_test.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	Counts []int  `json:"count,omitempty"`
+}
+type ResponseData struct {
+	All []Data `json:"all,omitempty"`
+}
+
+func InitData(t *testing.T) {
+	dg, err := testutil.DgraphClient(testutil.SockAddr)
+	testutil.DropAll(t, dg)
+	require.NoError(t, err)
```
Comment (jarifibrahim) [inline systest/ludicrous/upsert_test.go:32]: Check error before using the `dg`.

File: systest/ludicrous/upsert_test.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
+package worker
```
Comment (jarifibrahim) [inline systest/ludicrous/upsert_test.go:1]: This should be `main`. Also, the license is missing.

File: systest/ludicrous/upsert_test.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+	ctx := context.Background()
+	dg, err := testutil.DgraphClient(testutil.SockAddr)
+	require.NoError(t, err)
+
+	count := 10
+	var wg sync.WaitGroup
+	wg.Add(count)
+	mutation := func(i int) {
+		defer wg.Done()
+		query := fmt.Sprintf(`query {
```
Comment (jarifibrahim) [inline systest/ludicrous/upsert_test.go:73]: query := ` .... ` would also work. You don't need the `fmt.Sprintf`

**Review by jarifibrahim — APPROVED** [review]
Some minor comments. Good work @rahulgurnani . This was a hairy bug!

File: systest/ludicrous/upsert_test.go (hunk 3 above)
Comment (rahulgurnani) [reply systest/ludicrous/upsert_test.go:45]: Done, thanks!

File: systest/ludicrous/upsert_test.go (hunk 5 above)
Comment (rahulgurnani) [reply systest/ludicrous/upsert_test.go]: Done, thanks!

**Pull request merged**